docs/data-integrations/sources/adapt-existing-source.mdx

```diff
@@ -0,0 +1,218 @@
+                                password=self.config['password'],
+                                private_key_file=self.config.get(
+                                    'private_key_file'),
+                                port=self.config['port'])
+        client.close()
+```
+
+### Overwrite `write_schema` method
+
+Finally, we need to overwrite the `write_schema` method in the `tap_your_tap/sync.py` file. Add the following import:
+
+`from mage_integrations.sources.messages import write_schema`
+
+Now, switch out the default function (likely `singer.write_schema`) for the mage function using the default arguments. Here's an example:
+
+```python
+  write_schema(
+        stream_name=catalog['tap_stream_id'],
+        schema=schema,
+        key_properties=catalog.get('key_properties', ['id']),
+        bookmark_properties=catalog.get('bookmark_properties'),
+        disable_column_type_check=catalog.get('disable_column_type_check'),
+        partition_keys=catalog.get('partition_keys'),
+        replication_method=catalog.get('replication_method'),
+        stream_alias=catalog.get('stream_alias'),
+        unique_conflict_method=catalog.get('unique_conflict_method'),
+        unique_constraints=catalog.get('unique_constraints'),
+    )
+```
+
+### Add logging
+
+To add verbose logging to Mage, add a logger argument each tap file, `tap_github/sync.py`, `tap_github/discover.py`:
+
+```python
+def discover(client, logger=None):
+    """
+    Run the discovery mode, prepare the catalog file and return catalog.
+    """
+    mage_logger = logger or LOGGER
+```
+
+Then replace usage of `LOGGER` with `mage_logger`.
+
+```python
+
+for stream_name, schema_dict in schemas.items():
+        try:
+            schema = Schema.from_dict(schema_dict)
+            mdata = field_metadata[stream_name]
+        except Exception as err:
+            # These were LOGGER previously
+            mage_logger.error(err)
+            mage_logger.error('stream_name: %s', stream_name)
+            mage_logger.error('type schema_dict: %s', type(schema_dict))
+            raise err
+```
+
+This will enable Mage to better log Singer taps! 🥳
+
+## Add source to the Mage UI
+
+Add the new source to the `SOURCES` list constant in [this](https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_integrations/sources/constants.py) file
+
+This will make your new source visible in the Mage UI.
+
+## Test your source
+
+Sources should be tested both in the console and via the UI. Follow the directions [here](https://github.com/mage-ai/mage-ai/blob/master/mage_integrations/README.md#testing-sync-locally) to configure a lightweight test of your sources taps via a bash terminal and the Mage UI in the dev docker instance. This entails:
+
+- Setting up a sample source and connection locally.
+- Running the tap directly, via Python.
+- Writing to an intermediate file to inspect the output.
+- Performing an end-to-end sync.
+
```

Comment:
Can we also add the checklist of testing through UI?

Reply:
The preceding paragraph indicates testing should be performed through the console _and_ UI.